lib/spack/spack/test/link_tree.py

```diff
@@ -38,6 +38,8 @@ class LinkTreeTest(unittest.TestCase):
 
     def setUp(self):
         self.stage = Stage('link-tree-test')
+        # FIXME : possibly this test needs to be refactored to avoid the explicit call to __enter__ and __exit__
```

Comment:
Why not move the logic in `__enter__` and `__exit__` into more appropriately named functions, then make `__enter__` and `__exit__` call them?  This would mirror Python's builtin file objects, which expose `open` and `close` but wrap them with `__enter__` and `__exit__`.